File: BJ/25304/main.py
bill = int(input())
N = int(input())

recipe = 0

for i in range(N):
    parse = input().split(" ")
    price , amount = int(parse[0]), int(parse[1])
    
    recipe = recipe + (price * amount)

if bill == recipe:
    print("Yes")
else:
    print("No")
    
    

File: BJ/25304/main.cpp
#include <iostream>

using namespace std;

int main(int argc, char** argv){
    int bill, N;
    int price = 0;

    cin >> bill >> N;
    for(int i = 0; i < N; i++){
        int tmp, amount;
        cin >> tmp >> amount;
        price += tmp * amount;
    }

    if (price != bill){
        cout << "No" << endl;
    }else{
        cout << "Yes" << endl;
    }
}